lib/srv/desktop/rdp/rdpclient/src/vchan.rs

```diff
@@ -26,25 +26,33 @@ use std::io::{Cursor, Read};
 /// (or a single chunk) over a virtual channel.
 /// See https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/343e4888-4c48-4054-b0e3-4e0762d1993c
 /// for more information about chunks.
+///
+/// To prevent memory DoS attacks that could take down the entire system running the Windows Desktop Service,
+/// Client drops all RDP messages > capacity.
 pub struct Client {
+    // capacity is effectively the maximum size of an RDP
+    // message we will receive from the RDP server, minus any
+    // ChannelPDUHeaders.
+    capacity: usize,
     data: Vec<u8>,
-}
-
-impl Default for Client {
-    fn default() -> Self {
-        Self::new()
-    }
+    drop_this_message: bool,
```

Comment:
This seems like an odd name for a field. What message is "this message" supposed to be referring to?

Reply:
I changed it to `drop_current_message`. It's a flag that keeps track of whether or not we've gone over `capacity` for the RDP message currently being read. We need some state to keep track of this, because `Client.read()` might be called several times for a single RDP message ("`current_message`").